docs/src/proposals/bidirectional_quic_reply.md

```diff
@@ -0,0 +1,101 @@
+in time. 
+
+As Solana is being developed fast and we have multiple development roadmaps to improve the scheduling
+stage of the cluster. I propose that we add a new notification mechanism to the Solana cluster
+to understand more about the transaction lifecycle. 
+This will make it easier to develop, integrate, scale, and understand the Solana cluster.
+
+## Solution
+Solana currently uses QUIC to receive transactions from different clients either other validators or 
+rpc nodes. Validator usually listens in the cluster using a unidirectional channel, i.e only clients can 
+send data to the cluster server and cannot reply on the same channel. I propose that we also add a 
+possibility for the server to listen to a bidirectional quic connection requests. When a server gets a
+request for this bidirectional quic channel it will get the transactions sent by the client using a 
+recv stream and the send stream will be used by a special class "bidirectional reply handler".
+
+The bidirectional quic reply handler will then keep track of the transactions it receives from the IP
+address which initialized the bidirectional channel. Similar to log messages we can emit notifications
+for each transaction signature to the bidirectional reply handler which will then forward the 
+notification back to the client using send stream. The client has to just initialize a 
+bidirectional channel once and keep it alive to receive notifications about the transactions that it 
+has sent to the cluster. If the connection will be kept open by the server for a predetermined N 
+seconds and then will be closed. We can emit these notification at the most important stages of banking
+stage and where transactions are probably lost.
+
+A TPU client which connects to FANOUT_SLOTS/NUMBER_OF_SLOTS_PER_LEADER number of the next leader 
+then will initialize a bidirectional channel once with each of the leaders and send the transactions. 
+Each leader will then reply how it responded to the transactions and we can easily determine
+the lifecycle of the transaction. We add a mechanism in the TPU client called a bidirectional
+reply handler which will take care of the notifications received by each leader, and forward the 
+notifications to a crossbeam channel to be processed by the user of the TPU client.
+
+To avoid loops and over-notifications, each bidirectional channel will be limited to a single validator,
+the validator will not create bidirectional channels with other validators to get notifications.
+The size of the reply message will be a fixed length of N bytes so that the client will exactly 
+know that it has received a notification from the server after receiving N bytes. So the length of the 
+notification is limited to 128 bytes maximum. Here is the proposed structure:
+
+```
+pub struct QuicReplyMessage {
+    sender_identity: Pubkey,
+    transaction_signature: Signature,
+    message: [u8; 128],
```

Comment:
Actually we want to keep the length of the structure fixed so that when we decode on receiver side we know we got a new message after we read fixed bytes from the quic channel.